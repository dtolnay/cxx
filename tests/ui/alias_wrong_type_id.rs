#[cxx::bridge(namespace = folly)]
mod here {
    extern "C" {
        type StringPiece;
    }
}

#[cxx::bridge(namespace = folly)]
mod there {
    extern "C" {
        type ByteRange = crate::here::StringPiece;
    }
}

fn main() {}
