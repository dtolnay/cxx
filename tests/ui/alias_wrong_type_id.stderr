error[E0271]: type mismatch resolving `<StringPiece as ExternType>::Id == (f, o, l, l, y, (), B, y, t, e, R, a, n, g, e)`
   --> $DIR/alias_wrong_type_id.rs:11:9
    |
11  |         type ByteRange = crate::here::StringPiece;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected a tuple with 15 elements, found one with 17 elements
    |
   ::: $WORKSPACE/src/extern_type.rs
    |
    | pub fn verify_extern_type<T: ExternType<Kind = Kind, Id = Id>, Kind, Id>() {}
    |                                                      ------- required by this bound in `verify_extern_type`
    |
    = note: expected tuple `(f, o, l, l, y, (), B, y, t, e, R, a, n, g, e)`
               found tuple `(f, o, l, l, y, (), S, t, r, i, n, g, P, i, e, c, e)`
